app/Http/Controllers/Assets/AssetsController.php

```diff
@@ -478,9 +478,14 @@ public function getAssetByTag(Request $request, $tag=null) : RedirectResponse
         $tag = $tag ? $tag : $request->get('assetTag');
         $topsearch = ($request->get('topsearch') == 'true');
 
-        if (! $asset = Asset::where('asset_tag', '=', $tag)->first()) {
-            return redirect()->route('hardware.index')->with('error', trans('admin/hardware/message.does_not_exist'));
+        // Search for an exact and unique asset tag match
+        $assets = Asset::where('asset_tag', '=', $tag);
+
+        // If not a unique result, redirect to the index view
+        if ($assets->count() != 1) {
+            return redirect()->route('hardware.index')->with('search', $tag);
```

Comment:
I think we'd need to return this with a message of some kind - just returning them to the index with no explanation seems like it would be confusing.

Reply:
Throwing an "error" will likely be too aggressive since users may use the search field for other values than asset tag. A soft "info" may be more appropriate, with the asset tag included, to ensure users are aware of why they get this.

I tried to stay aligned with available localization, but perhaps the message should be more explicit.

![image](https://github.com/user-attachments/assets/ba71d68b-b73b-4b68-9d0b-f691e804938a)

Reply:
That makes sense to me - though I might make it a warning instead of info. I'm concerned that pre-populating that search might be confusing though, since there are results there. (The text does specify "with asset tag Laptop", but I bet some people might get confused.

Reply:
Result with the last commit ad79424
![image](https://github.com/user-attachments/assets/7d26c3f0-cbcf-498b-b9bc-d0bbdcf3554e)